tests/test_util.py

```diff
@@ -402,25 +421,38 @@ def request_callback(request):
 
                 digest = 'v1-converted-digest'
 
-        headers = {
-            'Content-Type': '{}+jsonish'.format(media_type_prefix),
-            'Docker-Content-Digest': digest
-        }
+        headers = {}
+        if digest_is_v1 or has_content_type_header:
+            headers['Content-Type'] = '{}+jsonish'.format(media_type_prefix)
+        if digest_is_v1 or has_content_digest:
+            headers['Docker-Content-Digest'] = digest
+
         return (200, headers, '')
 
     responses.add_callback(responses.GET, url, callback=request_callback)
 
-    actual_digests = get_manifest_digests(**kwargs)
+    if not digest_is_v1 and not has_content_type_header:
+        with pytest.raises(RuntimeError):
+            get_manifest_digests(**kwargs)
+        return
+    else:
+        actual_digests = get_manifest_digests(**kwargs)
 
     if digest_is_v1:
         assert actual_digests.v1 == 'v1-digest'
-        assert actual_digests.v2 is None
+        assert actual_digests.v2 in [True, None]
```

Comment:
`digests.v2` would return `None` if registry/pulp don't support v2 schema 2 images at all. It would return `True` if it does support, but the image could not be converted, and finally it would return the digest string if the digest is in the headers/manifest.

The tests should be rewritten to have this tested for each specific case though